src/domain/products/product-form/form/product-form-context.tsx

```diff
@@ -45,20 +65,25 @@ export const ProductFormProvider = ({
   const [images, setImages] = React.useState<any[]>([])
   const [variants, setVariants] = React.useState<any[]>([])
   const [productOptions, setProductOptions] = React.useState<any[]>([])
+  const [hasImagesChanged, setHasImagesChanged] = React.useState(false)
 
   const appendImage = (image) => setImages([...images, image])
 
   const removeImage = (image) => {
-    const idx = images.findIndex((img) => img.image === image.image)
-    if (idx !== -1) {
-      images.splice(idx, 1)
-    }
-    setImages([...images])
+    const tmp = images.filter((img) => img.url !== image.url)
+    setImages(tmp)
   }
-
   const methods = useForm()
 
   React.useEffect(() => {
+    if (JSON.stringify(product.images) !== JSON.stringify(images)) {
+      setHasImagesChanged(true)
+    } else {
+      setHasImagesChanged(false)
+    }
+  }, [JSON.stringify(images)])
```

Comment:
suggestion: I think we could handle `images` change detection in a less "expensive" way without using `JSON.stringify()`. Wdyt about removing this `useEffect` and adding `setHasImagesChanged(true)` call to `appendImage/removeImage` and also setting the flag to `false` in the `resetForm`.

Reply:
Agree that it would be nice to use a less expensive approach, but the issue with relying on append/remove is that the design allows you to reorganise image order, which would not be detected by calling these methods. But at the same time we don't actually support ordering images in the backend, so will have to check in if it's something that we are going to support at some point :)

Reply:
Think it's a good suggestion @fPolic. We will most likely build out image ordering on products as part of the Extend Product API epic, that we have coming up, but until then I think we should remove both the comparison here and the ability to drag images. Let me know, if this works @kasperkristensen. If so, I'll create a ticket for the latter

Reply:
@olivermrbl Definetly makes sense to remove the re-ordering as long as it is not supported. I'll remove the comparison in this PR, and then we will open a seperate PR to remove the dragable table? :)

Reply:
Yes, sounds good!